11-Backbonejs/06-Adding-Custom-Events.md

```diff
@@ -0,0 +1,257 @@
+
+The `trigger` function causes all callbacks to the given Event to run.  You can specify an argument to pass to the callback function as a 2nd argument.  In the example above we did so here:  `task.trigger("Past_Due", {msg: "past due",
+                            date: new Date()                        
+});`.
+
+
+
+## Adding Events To A View
+
+There are two types of Events a View can listen for in Backbone.  The first are DOM events like we've dealt with earlier.  The second are custom events using Backbone's Event API like we've seen above.  
+
+DOM events are bound using the View's `events` property or using `jQuery.on()`.  Look at [this CodePen](http://codepen.io/CheezItMan/pen/GNvMVG?editors=1010).  In this CodePen we have created a view for a Task item and added an event when the view is clicked on.
+
+Dom Events are added as an Object property of the view with the name of the event as the "key" and the value being the name of the function which serves as the callback function for the event.  
+
+```javascript
+// 'app/views/task.js'
+
+var TaskView = Backbone.View.extend({
+  tagName: "li",
+  events: {
+    "click": "onClick"
+  },
+  onClick: function() {
+    console.log("Clicked!  Model:  " + this.model.get("title"));
+  },
+  render: function() {
+    this.$el.html(this.model.get("title"));
+    return this;
+  }
+});
+``` 
+
+If you wanted to listen for a specific item within the view you could, with jQuery, specify the item using a CSS Selector.  For example if there was a block within the view with a class `title` and you wanted to listen for clicks on that block, you could put this line into the events hash `"click .title": "onTitleClick"`.  This would cause the View to listen for clicks on any element within the view and with the `title` class.  When any sub-element with that class is clicked on the `onTitleClick` event handler will run.
+
+
+Backbone Events API allows you to add event handlers with the `.on` method as you saw earlier.  This lets you create custom events that can be triggered in your Models or Views to update other objects when something occurs, a change in status, user interaction etc.  
+
+Looking at the TodoViews in the Codepen, we can see when a task is clicked on using a DOM event.  However it would be REALLY useful if we could update another view when a Todo item is clicked on, for example to show more details on the selected Todo item.  Thus we will return to the Backbone Events API and the idea of an Event Bus later.
```

Comment:
I think in the Codepen we're using the name `TaskView` rather than `TodoView`. Also "Todo item" -> "Task item".